pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2466,29 +2462,47 @@ private String receiveCommandStatus() throws IOException {
   private void interpretCommandStatus(String status, ResultHandler handler) {
     long oid = 0;
     long count = 0;
-    Matcher matcher = COMMAND_COMPLETE_PATTERN.matcher(status);
-    if (matcher.matches()) {
-      // String command = matcher.group(1);
-      String group2 = matcher.group(2);
-      String group3 = matcher.group(3);
+
+    int marker1 = -1;
+    int marker2 = -1;
+    // This code processes the CommandComplete (B) message.
+    // Status is in the format of "COMMAND OID ROWS" where both 'OID' and 'ROWS' are optional
+    // and COMMAND can have spaces within it, like CREATE TABLE.
+    // Start with the simple case, and then proceed to scan backwards
+    if (status.indexOf(" ") != -1) {
```

Comment:
Does this check actually add value? In the case with only a command with no spaces, this requires examining each char. Going into the for loop below would stop looking at chars as soon as a non-digit is encountered.